common/cluster/archival_status.go

```diff
@@ -87,6 +82,16 @@ func (a *ArchivalConfig) EnableReadFromArchival() bool {
 	return a.enableReadFromArchival
 }
 
+func (a *ArchivalConfig) isDomainDefaultValid() bool {
+	if a.domainDefaultStatus == ArchivalPaused {
+		return false
+	}
+
+	URISet := len(a.domainDefaultURI) != 0
```

Comment:
I think we should also check if the URI is valid here. But before I do that, I think we should not make ValidateURI a method of an archiver instance. Instead if should be an exported function of the archiver package. Otherwise, whenever we want to validate a URI, we need to create an archiver instance. We can also consider add another exported method to the provider package to validate any given URI, regardless of their scheme.

Reply:
It's a good thought but may not be easy. Consider the case of S3 - to check if a bucket is valid you actually need to make a call to blobstore to see if it exists. In the case of filestore we can do validation just based on a regex of the URI but that is not true of all archiver implementations. I think its reasonable that validateURI is part of archiver methods and just rely on calling that in frontend whenever a domain is actually enabled for archival.